File: photovoltaicV2/src/app/services/models/construction.ts
/* tslint:disable */
/* eslint-disable */
import { Activity } from '../models/activity';
import { Amendment } from '../models/amendment';
import { Bill } from '../models/bill';
import { Company } from '../models/company';
import { Devis } from '../models/devis';
export interface Construction {
  activities?: Array<Activity>;
  amendment?: Array<Amendment>;
  billList?: Array<Bill>;
  company?: Company;
  customer?: number;
  description?: string;
  devisList?: Array<Devis>;
  id?: number;
  location?: string;
}

File: New BackEnd/demo/src/main/java/com/example/demo/models/Construction.java
package com.example.demo.models;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIdentityReference;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.Data;

import java.util.List;

@Data
@Entity
@Table(name = "construction")
public class Construction {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String description;
    private String location;
    @JsonIdentityInfo(generator= ObjectIdGenerators.PropertyGenerator.class, property="id")
    @JsonIdentityReference(alwaysAsId=true)
    @OneToMany(mappedBy = "construction")
    private List<Bill> billList;
    @OneToMany(mappedBy = "construction")
    private List<Activity> activities;
    @OneToMany(mappedBy = "construction")
    private List<Amendment> amendment;
    @JsonIdentityInfo(generator= ObjectIdGenerators.PropertyGenerator.class, property="id")
    @JsonIdentityReference(alwaysAsId=true)
    @OneToMany(mappedBy = "construction")
    private List<Devis> devisList;
    @JsonIdentityInfo(generator= ObjectIdGenerators.PropertyGenerator.class, property="id")
    @JsonIdentityReference(alwaysAsId=true)
    @ManyToOne
    private Customer customer;
    @ManyToOne
    private Company company;
}
